src/main/java/edu/harvard/iq/dataverse/engine/command/impl/MergeInAccountCommand.java

```diff
@@ -0,0 +1,170 @@
+        }
+
+        //DVObjects creator and release
+        for (DvObject dvo : ctxt.dvObjects().findByAuthenticatedUserId(consumedAU)) {
+            if (dvo.getCreator().equals(consumedAU)){
+                dvo.setCreator(ongoingAU);
+            }
+            if (dvo.getReleaseUser() != null &&  dvo.getReleaseUser().equals(consumedAU)){
+                dvo.setReleaseUser(ongoingAU);
+            }
+            ctxt.em().merge(dvo);
+        }
+        
+        //GuestbookResponse
+        for (GuestbookResponse gbr : ctxt.responses().findByAuthenticatedUserId(consumedAU)) {
+            gbr.setAuthenticatedUser(ongoingAU);
+            ctxt.em().merge(gbr);
+        }
+        
+        //UserNotification
+        for (UserNotification note : ctxt.notifications().findByUser(consumedAU.getId())) {
+            note.setUser(ongoingAU);
+            ctxt.em().merge(note);
+        }
+        
+        //SavedSearch
+        for (SavedSearch search : ctxt.savedSearches().findByAuthenticatedUser(consumedAU)) {
+            search.setCreator(ongoingAU);
+            ctxt.em().merge(search);
+        }
+        
+        //Workflow Comments
+        for (WorkflowComment wc : ctxt.authentication().getWorkflowCommentsByAuthenticatedUser(consumedAU)) {
+            wc.setAuthenticatedUser(ongoingAU);
+            ctxt.em().merge(wc);
+        }
+        
+        //ConfirmEmailData  
+        
+        ConfirmEmailData confirmEmailData = ctxt.confirmEmail().findSingleConfirmEmailDataByUser(consumedAU);       
+        ctxt.em().remove(confirmEmailData);
+        
+        //Access Request is not an entity. have to update with native query
+        
+        ctxt.em().createNativeQuery("UPDATE fileaccessrequests SET authenticated_user_id="+ongoingAU.getId()+" WHERE authenticated_user_id="+consumedAU.getId()).executeUpdate();
+        
+        ctxt.actionLog().changeUserIdentifierInHistory(consumedAU.getIdentifier(), ongoingAU.getIdentifier());
+        
+        //delete:
+        //  builtin user - if applicable
+        //  authenticated user
+        //  AuthenticatedUserLookup
+        //  apiToken
+        ApiToken toRemove = ctxt.authentication().findApiTokenByUser(consumedAU);
+        if(null != toRemove) { //not all users have apiTokens
+            ctxt.em().remove(toRemove);
+        }
+        AuthenticatedUserLookup consumedAUL = consumedAU.getAuthenticatedUserLookup();
+        ctxt.em().remove(consumedAUL);
+        ctxt.em().remove(consumedAU);
+        BuiltinUser consumedBuiltinUser = ctxt.builtinUsers().findByUserName(consumedAU.getUserIdentifier());
+        if (consumedBuiltinUser != null){
+            ctxt.em().remove(consumedBuiltinUser); 
+        }
+        
+        
+    }
+    
+    @Override
+    public String describe() {
+        return "User " + consumedAU.getUserIdentifier() + " ( type: " + consumedAU.getAuthenticatedUserLookup().getAuthenticationProviderId() + " ) merged into " +ongoingAU.getUserIdentifier();
```

Comment:
getAuthenticationProviderId is good but only provides "shib" and not more specific. We can either add in logic to parse the persistentuserid to get "Harvard IDP" OR maybe we just put in the whole persistentuserid (is that useful?) since that is onfo that would otherwise be lost?

@matthew-a-dunlap thoughts?

Reply:
I'm going to just add the whole persistentUserId, seems the easiest and most useful.